### PR TITLE
feat: terminfo parser

### PR DESCRIPTION
**Pull request by chivay**

### CODE REVIEW

File: terminal.cpp [hunk 1]
```diff
@@ -1,11 +1,124 @@
...
+  }
+
+  std::filesystem::path terminfo_base{"/usr/share/terminfo"};
+  auto terminfo_path{terminfo_base / term.substr(0, 1) / term};
+
+  return TermInfo::from_file(terminfo_path);
+}
+
+std::string_view TermInfo::query_string(const std::string_view &query)
+{
```
Comment (msm-code) [inline terminal.cpp]: is it clang-formatted? :thinking:

File: terminal.cpp [hunk 2]
```diff
@@ -1,11 +1,124 @@
...
+    term = "dumb";
+  }
+
+  std::filesystem::path terminfo_base{"/usr/share/terminfo"};
+  auto terminfo_path{terminfo_base / term.substr(0, 1) / term};
+
+  return TermInfo::from_file(terminfo_path);
+}
+
+std::string_view TermInfo::query_string(const std::string_view &query)
```
Comment (msm-code) [inline terminal.cpp]: pass std::string_view by value

File: terminal.cpp [hunk 3]
```diff
@@ -1,11 +1,124 @@
...
+{
+  return strings_[query];
+}
+
+TermInfo TermInfo::from_file(const std::filesystem::path &path) {
+  std::ifstream terminfo_file{path, std::ios::in | std::ios::binary};
+  enum class InfoType { Legacy, Extended };
+
+  int16_t magic{read_into<int16_t>(&terminfo_file)};
```
Comment (msm-code) [inline terminal.cpp:57]: maybe use auto here and below to avoid specifying the type twice? :thinking: But not important i guess.

**Review by msm-code — APPROVED** [review]
:potega:

File: terminal.cpp (hunk 1 above)
Comment (chivay) [reply terminal.cpp]: It is :thinking: `clang-format -i -style=Google *.cpp *.h`

File: terminal.cpp (hunk 2 above)
Comment (chivay) [reply terminal.cpp]: :thinking: true

File: terminal.cpp (hunk 3 above)
Comment (chivay) [reply terminal.cpp:57]: done

**Pull request merged**